lib/puppet/file_serving/metadata.rb

```diff
@@ -88,6 +90,21 @@ def collect_stat(path)
   def collect(source_permissions = nil)
     real_path = full_path
 
+    if static_catalog
+      if environment && environment.static_catalogs?
+        env_path = File.join(Puppet[:environmentpath], environment.name.to_s) # environment.name is a symbol sometimes
+        puts "Checking if #{real_path} is in #{env_path}"
+        if real_path.start_with? env_path
```

Comment:
Another option might be to add an `inlineable?` class method to `Puppet::FileServing::Content` and `Puppet::FileServing::Metadata`, and then have the file server just call:

```
return nil unless model.inlineable?(request, path)
```

where `Content.inlineable?` always returns true, and `Metadata.inlineable?` has the logic you've implemented, but just as a class method?